site/en/tutorials/keras/Intro_to_RNN.ipynb

```diff
@@ -0,0 +1,441 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Keras recurrent layers can be run in two different modes: they return either the full sequences of successive outputs for each timestep (a 3D tensor of shape (batch_size, timesteps, output_features)), or return only the last output for each input sequence (a 2D tensor of shape (batch_size, output_features)). These two modes are controlled by the return_sequences constructor argument.\n",
+    "\n",
+    "Let's take a look at an example that uses a SimpleRNN layer and returns only the output at the last timestep:\n",
+    "\n",
+    "Before we add a simple RNN layer, let's first create a embedding layer.  An embedding layer organize the words with similar meanings into similar vectors. Comparing to one-hot encoding, word embedding is dense and relative low dimensional.  To create an embedding layer, we will pass two parameters, the first parameter is the maximum number of words in each vector, the second paramater is the number of vectors for the embedding layer.  In this example, the embedding layer splits the words into 32 vectors, each vector contains maximum 10000 words.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tensorflow.keras.models import Sequential\n",
+    "from tensorflow.keras.layers import Embedding, SimpleRNN\n",
+    "\n",
+    "model = Sequential()\n",
+    "model.add(Embedding(10000, 32))\n",
+    "model.add(SimpleRNN(32))\n",
+    "model.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following example returns the full state sequence."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = Sequential()\n",
+    "model.add(Embedding(10000, 32))\n",
+    "model.add(SimpleRNN(32, return_sequences=True))\n",
+    "model.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is sometimes useful to stack several recurrent layers one after the other in order to increase the representational power of a network. In such a setup, you have to get all intermediate layers to return full sequences:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = Sequential()\n",
+    "model.add(Embedding(10000, 32))\n",
+    "model.add(SimpleRNN(32, return_sequences=True))\n",
+    "model.add(SimpleRNN(32, return_sequences=True))\n",
+    "model.add(SimpleRNN(32, return_sequences=True))\n",
+    "model.add(SimpleRNN(32))  # This last layer only returns the last outputs.\n",
+    "model.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's try to use such a model on the IMDB movie review classification problem. First, let's preprocess the data:"
```

Comment:
First, let's load and preprocess the data. The preprocessing consists basically of padding the sentences so that all have the same length. If the RNN is dynamic as in [tf.nn.dynamic_rnn](https://www.tensorflow.org/api_docs/python/tf/nn/dynamic_rnn) this step is not needed.